### PR TITLE
feat(rf-commissioning): Implement SSA Calibration Phase

### PR DESCRIPTION
**Pull request by lisazacarias**
### Summary
This PR implements a fully functional SSA Calibration phase for the RF commissioning application, replacing the previous placeholder with a complete workflow that includes phase execution logic, UI controls, PV integration, and automated validation.

#### 1. **Data Model Updates** (`data_models.py`)
- **Replaced** legacy fields (`initial_drive`, `num_attempts`) with new fields:
  - `slope_new`: New calibration slope value (5 decimal precision)
  - `max_fwd_pwr_w`: Maximum forward power in Watts
  - `calibration_passed`: Boolean pass/fail status with custom display text
- **Removed** computed properties: `initial_drive_percent`, `drive_reduction`, `succeeded_first_try`
- **Retained** `max_drive_percent` and updated `is_complete` to check for `slope_new`
- **Updated** `passed` property to return `calibration_passed` value
- Updated display label from "SSA Characterization" to "SSA Calibration"

#### 2. **Phase Logic** (`phases/ssa_char.py`) - **NEW**
Implements `SSACharPhase` with a 6-step automated workflow:

1. **`verify_initial_state`**: Checks SSA is not mid-calibration and reads initial values
2. **`set_drive_max`**: Writes requested drive max to SSA
3. **`reset_and_power_on`**: Resets faults, powers on SSA, clears cavity interlocks
4. **`start_calibration`**: Triggers calibration scan via EPICS PV
5. **`wait_for_completion`**: Polls calibration status with timeout protection
6. **`validate_and_push`**: Validates results (forward power, slope tolerance) and auto-pushes new slope to cavity register

**Features:**
- Configurable limits via `SSACalLimits` dataclass
- Automatic retry logic for transient errors
- Pause/abort support throughout execution
- Dry-run mode for testing
- Comprehensive error handling with specific failure messages

#### 3. **Controller** (`ui/controllers/ssa_char_controller.py`) - **NEW**
Manages display logic, phase execution orchestration, and PV connections:

- **PV Management**: Auto-wires SSA PVs (slope, drive max, cal status, forward power)
- **Phase Execution**: 
  - Validates prerequisites before execution
  - Runs phase asynchronously in background thread
  - Tracks phase instance ID for database persistence
  - Auto-creates records if none active
- **User Actions**:
  - Manual "Push Slope" button to re-apply calibration
  - "Plot" button to display most recent calibration plot from filesystem
- **State Management**: Updates UI state (buttons, progress, status labels) throughout execution

#### 4. **UI Builder** (`ui/builders/phase_builders.py`)
Implements `SSACharUI` with two-panel layout:

**Left Panel:**
- Input controls (drive max spinbox, plot button)
- Phase history log
- Main toolbar (run/pause/abort)

**Right Panel:**
- Live PV readouts with 3-column grid:
  - **New** (calibration results)
  - **Current** (active values)
  - Labels: SSA Slope, Drive Max, Max Fwd Pwr, Cal Status
- Local phase status indicators
- Manual "Push New Slope to Cavity" button
- Stored data section (operator, timestamp, etc.)

**Styling:** Unified grid layout with color-coded headers, improved spacing/alignment

#### 5. **Display** (`ui/displays/ssa_char.py`) - **NEW**
`SSACharDisplay` integrates controller and UI:

- Connects controller signals (`phase_completed`, PV updates)
- Implements slot handlers for user actions
- Updates local result indicators (PASS/FAIL with color coding)
- Refreshes stored data readouts from record
- Notifies parent container on phase completion

#### 6. **SSA Utilities** (`utils/sc_linac/ssa.py`)
Added new PV addresses:
- `drive_max_new_pv`: New drive max from calibration
- `drive_max_current_pv`: Currently active drive max
- `saved_slope_pv`: Saved slope value

#### 7. **Platform Paths** (`utils/platform_paths.py`)
Added `get_ssa_cal_base_dir()` to locate calibration plot storage:
- Linux: `/u1/lcls/physics/rf_lcls2/ssa_cal`
- Dev: `~/ssa_cal`

#### 8. **Simulation Updates**
- `CavityPVGroup`: Implemented `push_ssa_slope` logic to copy `slope_new` → `slope_old`
- `SSAPVGroup`: Added `drive_max_new` and `drive_max_current` PVs, auto-populated during calibration

#### 9. **Testing**
Added comprehensive test coverage:
- **`test_ssa_char.py`**: 450+ lines covering all phase steps, edge cases, error paths
- **`test_ssa_char_controller.py`**: 981 lines covering controller logic, PV wiring, UI state, async execution
- **`test_phase_builders.py`**: UI widget registration and layout validation
- Updated existing tests for data model changes

#### 10. **Registration & Visibility**
- Added `SSACharPhase` to `phases/__init__.py`
- Added `SSACharUI` to `ui/builders/__init__.py`
- Updated `DEFAULT_BETA_VISIBLE_PHASES` to include `SSA_CHAR`
- Registered display in `ui/displays/registry.py`

### Migration Notes

**Breaking Changes:**
- `SSACharacterization` fields renamed/removed:
  - ❌ `initial_drive`, `num_attempts` → ✅ `slope_new`, `max_fwd_pwr_w`, `calibration_passed`
  - Computed properties `drive_reduction`, `succeeded_first_try` removed

**Database:** Existing records will need migration to populate new fields or handle `None` values gracefully.

### Testing Instructions

1. **Unit Tests**: Run `pytest tests/applications/rf_commissioning/phases/test_ssa_char.py -v`
2. **Simulation Mode**:
   ```bash
   # Terminal 1: Start IOC simulator
   python -m sc_linac_physics.utils.simulation.sc_linac_physics_service
   
   # Terminal 2: Launch UI
   pydm path/to/multi_phase_screen.py
   ```
3. **Workflow Validation**:
   - Select CM02, Cavity 1
   - Navigate to "SSA Calibration" tab
   - Set drive max (default 0.670)
   - Click "Run Calibration"
   - Verify all 6 steps execute
   - Check stored data updates with calibration results
   - Test pause/abort mid-execution
   - Click "Push New Slope to Cavity" to manually apply
   - Click "Plot" to view calibration curve (requires filesystem data)

### Screenshots/Demo

https://github.com/user-attachments/assets/5b26a5d1-b759-4817-b439-74b46060aa75

### Related Issues
- Closes #XXX (if applicable)
- Related to SSA calibration automation initiative

---

**Checklist:**
- [x] Unit tests added/updated
- [x] Integration with existing phase workflow
- [x] Documentation comments in code
- [x] Backward compatibility considered (migration notes provided)
- [x] Simulation service updated for testing

### CODE REVIEW

File: src/sc_linac_physics/applications/rf_commissioning/models/data_models.py [hunk 1]
```diff
@@ -282,18 +282,28 @@ class SSACharacterization:
...
-        widget_name="ssa_initial_drive",
-        source_attr="initial_drive_percent",
-        format_spec=".2f",
-        unit="%",
+        label="Slope",
+        widget_name="ssa_slope_new",
+        source_attr="slope_new",
+        format_spec=".5f",
+    )
+    max_fwd_pwr_w: float | None = phase_display_field(
```
Comment (s-aderhold) [inline src/sc_linac_physics/applications/rf_commissioning/models/data_models.py]: We haven't typically included the unit in variable names. Do you plan to do this for everything going forward?

File: src/sc_linac_physics/applications/rf_commissioning/phases/ssa_char.py [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+                result=PhaseResult.FAILED,
+                message="Calibration result not good (CALSTAT)",
+            )
+
+        max_fwd_pwr = ssa.max_fwd_pwr
+        if max_fwd_pwr < ssa.fwd_power_lower_limit:
+            return PhaseStepResult(
+                result=PhaseResult.FAILED,
+                message=(
+                    f"Forward power {max_fwd_pwr:.0f} W below limit "
```
Comment (s-aderhold) [inline src/sc_linac_physics/applications/rf_commissioning/phases/ssa_char.py]:
```suggestion
                    f"Forward power {max_fwd_pwr:.0f} W below limit of"
```

File: src/sc_linac_physics/applications/rf_commissioning/ui/controllers/ssa_char_controller.py [hunk 3]
```diff
@@ -0,0 +1,660 @@
...
+        except Exception as exc:
+            import traceback
+
+            self.view.show_error(f"Failed to start calibration: {exc}")
+            self.view.log_message(f"Traceback: {traceback.format_exc()}")
+
+    def _get_drive_max(self) -> float:
+        if hasattr(self.view, "drive_max_spinbox"):
+            return self.view.drive_max_spinbox.value()
+        return 0.670
```
Comment (s-aderhold) [inline src/sc_linac_physics/applications/rf_commissioning/ui/controllers/ssa_char_controller.py]: Are we okay with this being a magic number without comment?

**Review by s-aderhold — APPROVED** [review]

File: src/sc_linac_physics/applications/rf_commissioning/ui/controllers/ssa_char_controller.py (hunk 3 above)
Comment (lisazacarias) [reply src/sc_linac_physics/applications/rf_commissioning/ui/controllers/ssa_char_controller.py]: I meant to replace it with a pydm spin box I just forgot
